### PR TITLE
feat: marimo export ipynb

### PR DESCRIPTION
**Pull request by akshayka**
Export marimo notebooks to Jupyter (ipynb) notebooks.

This is a simple implementation that doesn't convert markdown or create/save outputs. Still, I had a need for this today -- more convenient than copy-pasting, and didn't want to go through VSCode.

It requires an optional dependency to do the format (`nbformat`).

For #717

### CODE REVIEW

**Review by mscolnick — APPROVED** [review]

File: marimo/_server/export/exporter.py [hunk 1]
```diff
@@ -119,6 +120,25 @@ def export_as_script(
...
 
+    def export_as_ipynb(
+        self,
+        file_manager: AppFileManager,
+    ) -> tuple[str, str]:
+        import nbformat
+
+        notebook = nbformat.v4.new_notebook()  # type: ignore
+        graph = file_manager.app.graph
+        notebook["cells"] = [
```
Comment (dmadisetti) [inline marimo/_server/export/exporter.py:131]: You could use `exporter/utils.is_markdown` to set markdown cells

File: tests/_cli/test_cli_export.py [hunk 2]
```diff
@@ -434,3 +434,18 @@ def test_export_watch_script_no_out_dir(
...
+
+
+class TestExportIpynb:
+    @pytest.mark.skipif(
+        not DependencyManager.has_nbformat(),
+        reason="This test requires nbformat.",
+    )
+    def test_export_ipynb(self, temp_marimo_file: str) -> None:
+        p = subprocess.run(
+            ["marimo", "export", "md", temp_marimo_file],
```
Comment (dmadisetti) [inline tests/_cli/test_cli_export.py:446]: `["marimo", "export", "ipynb", temp_marimo_file]` ?

Comment (akshayka) [reply tests/_cli/test_cli_export.py:446]: Oops, thanks

File: marimo/_server/export/exporter.py (hunk 1 above)
Comment (akshayka) [reply marimo/_server/export/exporter.py:141]: oh sweet, done -- thanks

File: tests/_cli/snapshots/ipynb.txt [hunk 3]
```diff
@@ -3,35 +3,35 @@
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "0e29fba8",
+   "id": "cd2168d6",
```
Comment (dmadisetti) [inline tests/_cli/snapshots/ipynb.txt:6]: Non-deterministic so will fail again

I think in cell creation you can pass in "id" as a kwarg- could just use the marimo cell_id, I think it's loose

**Review by mscolnick — APPROVED** [review]

**Pull request merged**